microSALT/utils/pubmlst/get_credentials.py

```diff
@@ -66,23 +68,68 @@ def save_to_credentials_py(client_id, client_secret, access_token, access_secret
     print(f"Tokens saved to {credentials_file}")
 
 
-def main():
+def main(service, species=None):
     try:
-        pubmlst_config = app.config["pubmlst"]
-        client_id = pubmlst_config["client_id"]
-        client_secret = pubmlst_config["client_secret"]
+        service_config = get_service_config(service)
+        bigsd_config = service_config["config"]
+        client_id = bigsd_config["client_id"]
+        client_secret = bigsd_config["client_secret"]
         validate_credentials(client_id, client_secret)
-        credentials_path = get_path(folders_config, credentials_path_key)
-        credentials_file = os.path.join(credentials_path, pubmlst_auth_credentials_file_name)
-        access_token, access_secret = get_new_access_token(client_id, client_secret)
+
+        # Determine the database
+        if service == "pubmlst":
+            db = service_config["database"]
+        elif service == "pasteur":
+            if not species:
+                raise ValueError("For the 'pasteur' service, you must provide a species.")
+            db = f"pubmlst_{species}_seqdef"
+        else:
+            raise ValueError(f"Unknown service: {service}")
+
+        credentials_path = get_path(folders_config, CREDENTIALS_KEY)
+        credentials_file = os.path.join(
+            credentials_path, service_config.get("auth_credentials_file_name")
+        )
+
+        access_token, access_secret = get_new_access_token(
+            client_id=client_id,
+            client_secret=client_secret,
+            db=db,
+            base_api=service_config["base_api"],
+            base_web=service_config["base_web"],
+        )
+
         print(f"\nAccess Token: {access_token}")
         print(f"Access Token Secret: {access_secret}")
-        save_to_credentials_py(client_id, client_secret, access_token, access_secret, credentials_path, credentials_file)
+
+        save_to_credentials_py(
+            client_id,
+            client_secret,
+            access_token,
+            access_secret,
+            credentials_path,
+            credentials_file,
+        )
 
     except Exception as e:
         print(f"Error: {e}")
         sys.exit(1)
 
 
 if __name__ == "__main__":
-    main()
+    parser = ArgumentParser(description="Get PUBMLST or Pasteur credentials.")
+    parser.add_argument(
+        "-s",
+        "--service",
+        type=str,
+        default="pubmlst",
+        help="Service name (default: pubmlst)",
+    )
+    parser.add_argument(
+        "-sp",
```

Comment:
Isn't it a little confusing to use multi-character short-hand flags, since `-sp` would typically be interpreted as the same as `-s -p`?